src/3rd_party_update.c

```diff
@@ -299,34 +335,8 @@ enum swupd_code third_party_update_main(int argc, char **argv)
 		steps_in_update = 11;
 	}
 
-	/* update 3rd-party bundles */
-	ret_code = third_party_run_operation_multirepo(cmdline_option_repo, update_repos, SWUPD_NO, "update", steps_in_update);
-	if (ret_code) {
-		goto exit;
-	}
-
-	/* read the current template copy */
-	template_file = sys_path_join("%s/%s/%s", globals_bkp.path_prefix, SWUPD_3RD_PARTY_DIR, SWUPD_3RD_PARTY_TEMPLATE_FILE);
-	template = sys_mmap_file(template_file, &template_len);
-
-	if (!template || strncmp(template, SCRIPT_TEMPLATE, template_len) != 0) {
-		/* there is no template file, or the template changed,
-		 * all scripts need to be recreated */
-		info("The scripts that export binaries from 3rd-party repositories need to be regenerated\n\n");
-		ret_code = third_party_run_operation_multirepo(NULL, regenerate_all_wrapper_scripts, SWUPD_OK, "regenerate_scripts", steps_in_update);
+	ret_code = third_party_execute_update(steps_in_update);
```

Comment:
This looks incorrect. Why do we need the number of steps from update in the 3rd-party update?